### PR TITLE
Extract user location

### PR DESCRIPTION
**Pull request by ravifrancesco**
Added classes UserPoint, GPSTracker, and UserLocationHandler.
Added GUI for demoing these features as well as the distance calculation.

### CODE REVIEW

**Review by giommok — APPROVED** [review]
All good

File: code/app/src/main/java/com/github/ravifrancesco/softwaredevproject/GPSTracker.java [hunk 1]
```diff
@@ -0,0 +1,193 @@
...
+    }
+
+    // returns a boolean that indicates if the permits for accessing location are ok
+    public boolean canGetLocation() {
+        return this.canGetLocation;
+    }
+
+    @Override
+    public void onLocationChanged(Location location) {
+    }
```
Comment (bgabriel998) [inline code/app/src/main/java/com/github/ravifrancesco/softwaredevproject/GPSTracker.java:174]: Don't we want to use the onLocationChanged callback to update the lat.long.alt. fields automatically ?

**Review by bgabriel998 — CHANGES_REQUESTED** [review]

File: code/app/src/main/java/com/github/ravifrancesco/softwaredevproject/GPSTracker.java (hunk 1 above)
Comment (ravifrancesco) [reply code/app/src/main/java/com/github/ravifrancesco/softwaredevproject/GPSTracker.java:174]: You're right! I'll try to do some modifications

**Review by bgabriel998 — APPROVED** [review]
Approved

**Comment by ravifrancesco** [comment]
Outdated

**Pull request closed without merging**